05.ls/ls_command.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+MAXIMUM_COLUMN = 3
+
+def main
+  path = if ARGV.size >= 1
+           File.join(ARGV[0], '*')
+         else
+           '*'
+         end
+  files = Dir.glob(path)
+  total_files_count = files.size
+  column_count = (total_files_count.to_f / MAXIMUM_COLUMN).ceil
+  columns = files.each_slice(column_count).to_a
+
+  # 配列ごとの最後に何もなかった場合'nil'を入れる
+  if columns.size >= MAXIMUM_COLUMN && total_files_count % MAXIMUM_COLUMN != 0
+    (MAXIMUM_COLUMN - total_files_count % MAXIMUM_COLUMN).to_i.times do
+      columns[-1] << nil
+    end
+  end
+  show_files(columns)
+end
+
+def show_files(columns)
+  longest_name = columns.flatten.max_by(&:size)
+  columns.transpose.each do |file_tables|
+    file_tables.each do |file_table|
```

Comment:
前回の僕のコメントの意図がうまく伝わってなかったようなので、具体的なコード例を書いておきます。

行と列で構成されるネストした配列がテーブル、というイメージなので、tableはメソッドの引数になります。
最後のeachは`columns.each do |column|`でもいいのですが、columnよりもさらに具体的な`file_paths.each do |file_path|`の方が可読性が上がるかもしれません。

```suggestion
def show_files(file_table)
  longest_name = columns.flatten.max_by(&:size)
  file_table.transpose.each do |file_paths|
    file_paths.each do |file_path|
```